src/app/services/courses/course.service.ts

```diff
@@ -6,45 +6,63 @@ import { Course } from '../../shared/models/course.model';
+        let courseArray = [
+            {id: 1, title: 'Video course 1', duration: 88, date: new Date(2018, 1, 12), topRated: false, description: 'Ante laborum massa, suspendisse sed adipiscing lectus magna, congue consectetuer aliquam quis magna tincidunt animi, lacus faucibus aliquam sed, neque wisi ac in arcu mi velit. In sit neque nibh faucibus vel. Felis eget id sed at. Eget mauris. Metus morbi quis consectetuer dictumst, pellentesque non, eu lectus non feugiat vel elementum, non commodo duis, magna accumsan.'},
+            {id: 2, title: 'Video course 2', duration: 15, date: new Date(2017, 12, 2), topRated: true, description: 'Ante laborum massa, suspendisse sed adipiscing lectus magna, congue consectetuer aliquam quis magna tincidunt animi, lacus faucibus aliquam sed, neque wisi ac in arcu mi velit. In sit neque nibh faucibus vel. Felis eget id sed at. Eget mauris. Metus morbi quis consectetuer dictumst, pellentesque non, eu lectus non feugiat vel elementum, non commodo duis, magna accumsan.'},
+            {id: 3, title: 'Video course 3', duration: 135, date: new Date(2017, 12, 17), topRated: false, description: 'In dictum ultricies sed. Massa elit posuere sit scelerisque, nullam commodi, wisi litora interdum tortor. Nulla quam rutrum, mollis odio wisi ac tempus, vehicula wisi tristique eget maecenas, suspendisse et neque elit nonummy arcu congue.'},
+            {id: 4, title: 'Video course 4', duration: 45, date: new Date(2018, 1, 11), topRated: false, description: 'Est penatibus nulla proin eleifend integer nonummy, non consequuntur vestibulum donec nullam penatibus, sociis elementum morbi, consequat ut eget sociis. Integer lacinia ligula blandit vestibulum, commodo volutpat quam quis, pede suspendisse aliquam duis non lectus praesent, proin justo, pede sit sit nec. Ut donec pellentesque et cum, per faucibus ac eu.'}
         ];
+
+        this.courses = Observable.of(courseArray);
     }
 
-    public getAll(): Course[] {
-        return this.courses;
+    public getAll(): Observable<Course[]> {
+        return this.courses
+            .map((courses: Course[]) => courses
+                .filter((x: Course) => {
+                    return x.date.getDate() > (new Date().getDate() - 14);
+                }
+            ));
     }
 
-    public getById(id: number): Course {
-        return this.courses.find((x: Course) => x.id === id);
+    public getById(id: number): Observable<Course> {
+        return this.courses
+            .map((courses: Course[]) => courses.find((course: Course) => course.id === id));
     }
 
-    public create(course: Course): Course {
-        this.courses.push(course);
-        return course;
+    public create(course: Course): Observable<Course> {
+        this.courses
+            .map((courses: Course[]) => courses.push(course));
+
+        return Observable.of(course);
```

Comment:
incorrect approach to reactive programming. Reactive streams are called reactive because your behavior reacts on data changes. Currently you are trying to mutate initial data instead of triggering a new set of data. This is incorrect. If you decided to store data in Observable, consider using Subject instead and call `.next()` when a new version of data is available.